torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2068,52 +2068,145 @@ Tensor svd_backward(const std::vector<torch::autograd::Variable> &grads, const T
   return u_term + sigma_term + v_term;
 }
 
+// The implementation follows:
 // "An extended collection of matrix derivative results for forward and reverse mode algorithmic differentiation"
 // https://people.maths.ox.ac.uk/gilesm/files/NA-08-01.pdf
+// However, the reference does not cover the constraints on eigenvectors to have 1-norm.
+// See the details below.
 Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self,
-                    bool eigenvectors, const Tensor& lambda, const Tensor& v) {
-  // This gradient only works for real eigenvalues at the moment.
+                    bool eigenvectors, const Tensor& lambda, const Tensor& u) {
```

Comment:
I'd suggest renaming `lambda` to `eigenvalues` to avoid confusion

Reply:
Could be confusing because eigenvalues corresponds to the Boolean flag computing the eigenvalues.

Reply:
Then maybe renaming `u` would also make sense? What about `eigvals/eigvars`?

Reply:
current schema decl in `derivatives.yaml` --> `eig_backward(grads, self, eigenvectors, eigenvalues, eigenvectors_return)`
I agree that we should rename `u` as well. I think this could be a good rename for the function inputs:
`Tensor eig_backward(const std::vector<torch::autograd::Variable> &grads, const Tensor& self, bool is_eigvec_tensor_nonempty, const Tensor& eigenvalues, const Tensor& eigenvectors) {`